dashboard/src/reducers/charts.ts

```diff
@@ -56,14 +59,20 @@ const chartsReducer = (
 ): IChartState => {
   switch (action.type) {
     case getType(actions.charts.requestCharts):
-      return { ...state, isFetching: true };
+      state.records.set(action.payload, false);
+      return { ...state, isFetching: true, records: state.records };
     case getType(actions.charts.requestChartsCategories):
       return { ...state, isFetching: true };
     case getType(actions.charts.receiveCharts):
+      state.records.set(state.page, true);
+      state.records.set(state.page + 1, false);
```

Comment:
do you need this line? Isn't the default `false`?

Reply:
I think so; when there is no `key`, the `map.get(xxx)` will return `undefined` instead of `false` (and this behavior is being used in https://github.com/kubeapps/kubeapps/pull/2264#discussion_r561066033)